### ISSUE TITLE
Build failure

### ISSUE THREAD

**Issue by mestery** [ipdk-io/ipdk#85]
With the latest code up to commit 52e06c4, when I run `ipdk build --no-cache` I see the following obtuse error and it fails:

```
-- Build files have been written to: /root/P4OVS_DEPS_SRC_CODE/gflags/build
make: *** empty string invalid as file name.  Stop.
The command '/bin/bash -c /root/start_p4ovs.sh /root &&      /root/scripts/run_cleanup.sh /root ${KEEP_SOURCE_CODE}' returned a non-zero code: 2
/git/ipdk/build
vagrant@ubuntu2004:/git/ipdk/build$
```

**Comment by mestery** [comment]
I am using Ubuntu 20.04, and my `ipdk.env` file looks as follows:

```
#
# IPDK CLI configuration file
# See for more information https://github.com/ipdk-io/ipdk/blob/main/build/IPDK_Container/README_DOCKER.md#CLI-configuration-settings
#
# /git/ipdk/build/scripts contains the location of the 'sourcing' ipdk script when read.
#
# When using ubuntu
BASE_IMG=ubuntu:20.04
IMAGE_NAME=ipdk/p4-ovs-ubuntu20.04
DOCKERFILE=${SCRIPT_DIR}/../Dockerfile.ubuntu
```

**Comment by mestery** [comment]
Even with a brand-new VM created, I still am hitting this build error.

@stolsma are you seeing this problem by chance? Note I'm using Ubuntu 20.04, will try 18.04 next.

**Comment by mestery** [comment]
Same issue with Ubuntu 18.04:

```
#
# IPDK CLI configuration file
# See for more information https://github.com/ipdk-io/ipdk/blob/main/build/IPDK_Container/README_DOCKER.md#CLI-configuration-settings
#
# /git/ipdk/build/scripts contains the location of the 'sourcing' ipdk script when read.
#

# When using ubuntu20.04
BASE_IMG=ubuntu:18.04
IMAGE_NAME=ipdk/p4-ovs-ubuntu18.04
DOCKERFILE=${SCRIPT_DIR}/../Dockerfile.ubuntu
```

**Comment by nupuruttarwar** [comment]
I am able to build fine on both Ubuntu 20.04 as well as 18.04 container. You will face these issues when there is an empty string in front of make. What's NUM_THREADS on your system? Can you remove quotes around NUM_THREADS after make in install_deps scriupt and see if this error goes away?

**Comment by mestery** [comment]
@nupuruttarwar Are you using the `ipdk` script to do the build, e.g. `ipdk build`? I can't stop the build to make that change. Regardless, I've pushed a branch to OVS to cleanup the `os_get_details.sh` script, as it's out of sync with the one in IPDK. Lets see if that fixes this. I'd still like to understand how you can get this to build using the `ipdk build` command however.

**Comment by nupuruttarwar** [comment]
yes, using ./ipdk build --no-cache --use-proxy. You can modify start_p4ovs.sh to comment out steps starting installation of dependencies and then execute build command. Like this:

if [ -z "${INSTALL_DEPENDENCIES}" ] || [ "${INSTALL_DEPENDENCIES}" == "y" ]
then
    get_p4ovs_repo
    build_p4sde
#    install_dependencies
#    build_p4c
fi
#build_p4ovs

**Comment by mestery** [comment]
This fixes the issue: https://github.com/ipdk-io/ovs/pull/17

**Comment by mestery** [comment]
After discussing with @nupuruttarwar, the problem here is the old version of `os_ver_details.sh` in the OVS repo only worked if you had > 4 CPU cores. This was fixed in the IPDK repository with #28. ipdk-io/ovs#17 will fix this in the OVS repository.
